internal/fleet/output_resource.go

```diff
@@ -174,6 +203,18 @@ func resourceOutputCreateLogstash(ctx context.Context, d *schema.ResourceData, m
 	if value, ok := d.Get("ca_trusted_fingerprint").(string); ok && value != "" {
 		reqData.CaTrustedFingerprint = &value
 	}
+	if value, ok := d.GetOk("ssl"); ok {
+		ssl := value.([]interface{})[0].(map[string]interface{})
```

Comment:
reqData.Ssl will be nil for all these assignments. You probably just want to set it to an empty struct here.